### DISCUSSION TITLE
TypeGrapghql Boilerplate CLI tool

### DISCUSSION THREAD

**Discussion by YassinEldeeb** [MichalLytek/type-graphql#1131] [Ideas]
Wouldn't it be nice to have a CLI tool like create-react-app but for setting up a new TypeGrapghql project.

It'll do all of that automatically with a little CLI tool with some useful tags to customize the setup wizard.
https://typegraphql.com/docs/installation.html

I'm open with contributing to get this feature, just making sure it's not already implemented.

**Comment by MichalLytek** [comment]
See #66.

Setup is different for every environment, like standalone app, lambda, next.js, etc.

**Reply by YassinEldeeb** [reply]
oh that does make sense.
but isn't the majority of people using type-graphql are standalone apps, so why not starting with adding support for a CLI tool to specifically setup type-graphql for standalone apps and then add simple instructions for the community to follow for contributing to add support for other environments later on, so it's a plus for the majority of people.

what do you think?

**Reply by YassinEldeeb** [reply]
I like to think of this problem the same way as writing models for typegraphql and ORMs using additional decorators so it's a single source of truth for the data in both the graphql-schema and for the database where type-graphql support most of the popular ones leaving others probably harder or specific use-cases for the community to implement like [prisma-tgql-types-gen](https://github.com/YassinEldeeb/prisma-tgql-types-gen).
